pkg/testing/runner/runner.go

```diff
@@ -192,43 +246,60 @@ func (r *Runner) Run(ctx context.Context) (Result, error) {
 
 // Clean performs a cleanup to ensure anything that could have been left running is removed.
 func (r *Runner) Clean() error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
-	defer cancel()
-	return r.ogcDown(ctx)
+	r.stateMx.Lock()
+	defer r.stateMx.Unlock()
```

Comment:
Clean is now only called separately from the actual running of the tests. It will always be called from a different process.

Reply:
Your not wrong the mutex could be removed here. I keep it hear so its clear that the `r.state` is a shared state and in the case that it was used from 2 goroutines it would be used correctly. It currently is not used from 2 goroutines, but when reading the code its clear that the state is protected by a lock.

So in short it could be removed, but I don't because of above.